src/Analyzers/Core/Analyzers/IDEDiagnosticIdToOptionMappingHelper.cs

```diff
@@ -28,6 +30,31 @@ public static bool TryGetMappedOptions(string diagnosticId, string language, out
                (s_diagnosticIdToLanguageSpecificOptionsMap.TryGetValue(language, out var map) &&
                 map.TryGetValue(diagnosticId, out options));
 
+        public static ImmutableArray<string> GetDiagnosticIdsForOption(IOption2 option, string language)
```

Comment:
Can we benefit from a BidirectionalMap here to avoid searching for keys by a value?

Reply:
BidirectionalMap is not appropriate for this case. there are many options per diagnostic id which is not a case handled by the current implementation.